### ISSUE TITLE
ccls/flymake false error reported

### ISSUE THREAD

**Issue by dlyr** [MaskRay/emacs-ccls#33]
So, I could not manage to make a minimal example.
Do not know if I do something wrong, feel free to ask other information.

Here is how to reproduce, with uptodate package, and master ccls (  https://github.com/MaskRay/ccls/commit/5c5bbb956ba72c073287e2771b3e799444e6cf77 ) run with  --log-file=/tmp/ccls.log --init='{"cache": {"directory": "/ssd/Builds/ccls-cache/"}}' 
`git clone --single-branch --branch ccls --recursive  https://github.com/dlyr/Radium-Engine/ TEST`
`cd TEST`
`mkdir build`
`cd build`
`cmake ../ -DCMAKE_EXPORT_COMPILE_COMMANDS=On -DRADIUM_USE_COTIRE=Off`
(not sure cotire interacts, but I use to turn it off with ccls)
`make -j`
 (not sure need to compile ... but there are some "installed" include and generated files ...)
`cd ../`
`ln -s build/compile_commands.json  .`
`emacs -Q config.el`
evaluate the lines here
open tests/CoreTests/src/algebra.cpp
wait everything loads, no flymake errors
add a blank line, save, and I get 19 flymake erros.

**Comment by MaskRay** [comment]
Can you dump requests/responses with https://github.com/MaskRay/ccls/wiki/Debugging#dump-lsp-requestsresponses ? Do you find `textDocument/publishDiagnostics`?

**Comment by dlyr** [comment]
Here is the log (start after emacs loaded, before saving)
there is a textDocument/publishDiagnostics seems to correspond to the reported error, 
but the project compiles and jump to definition works
[log.txt](https://github.com/MaskRay/emacs-ccls/files/2900429/log.txt)

**Comment by MaskRay** [comment]
If ccls sends `textDocument/publishDiagnostics`, the problem isn't on ccls' side. You may check if updating flymake works.

**Comment by dlyr** [comment]
well the error reported should not have been reported by ccls, since there are no error actually

**Comment by dlyr** [comment]
Is there any way I could find the command used by ccls to find that there is some error to report ?

**Comment by MaskRay** [comment]
https://github.com/MaskRay/ccls/wiki/Customization#-vn

**Comment by dlyr** [comment]
I still do not find the problem.
I have tracked the compile command, find that a compiler option I use is not supported by clang, removed it with "clang": {"excludeArgs": ["-Wno-misleading-indentation"]}} in ccls command options (by the way I have corrected the wiki to add the array definition, without it I got an error.

So the file compiles perfectly in command line with the same clang and arguments as shown in logs, but there still is some reported error by ccls.
As I understand when ccls send textDocument/publishDiagnostics it means that ccls found errors and report them to flymake ?
